### PR TITLE
Overridden start and due dates of problems 

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
Hi

In this PR I have overridden start and due dates of problems under units. This is to control due date of a unit from ccx schedule. Now check button on ccx course depends on due date of unit in schedule.

#100 
@pdpinch @giocalitri @pwilkins 
- CCX schedule with due date past.
screen shot 2015-10-21 at 7 55 20 pm

- CCX course with due date override. You can see there is no check button.
screen shot 2015-10-21 at 7 55 24 pm

- Master course still has check button because due date is not past yet.
screen shot 2015-10-21 at 7 55 31 pm

### CODE REVIEW

File: lms/djangoapps/ccx/views.py [hunk 1]
```diff
@@ -234,6 +234,16 @@ def override_fields(parent, data, graded, earliest=None, ccx_ids_to_delete=None)
                 graded[block.format] = graded.get(block.format, 0) + 1
 
             children = unit.get('children', None)
+            # For a vertical override start and due dates of its all problems.
+            if unit['category'] == u'vertical':
+                for component in block.get_children():
+                    if component.location.category == 'problem' and getattr(component, 'has_score', False):
```
Comment (giocalitri) [inline lms/djangoapps/ccx/views.py]: is `problem` supposed to be unicode too? (just asking)

**Comment by giocalitri** [comment]
just a small question, then I think it can be opened against edx

File: lms/djangoapps/ccx/views.py (hunk 1 above)
Comment (amir-qayyum-khan) [inline lms/djangoapps/ccx/views.py]: I dont think so, it is key. see https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/courses.py#L459

**Comment by amir-qayyum-khan** [comment]
@giocalitri ping me when you are done with reviewing and testing it on devstack. So that I can create PR on edX

**Comment by pdpinch** [comment]
Take a look at how IDDE handles this, since it has to override due dates as well.

**Comment by amir-qayyum-khan** [comment]
@pdpinch made change to cover all block those handle due and start date.
Going to look into IDDE

File: lms/djangoapps/ccx/views.py [hunk 2]
```diff
@@ -234,6 +234,17 @@ def override_fields(parent, data, graded, earliest=None, ccx_ids_to_delete=None)
                 graded[block.format] = graded.get(block.format, 0) + 1
 
             children = unit.get('children', None)
+            # For a vertical override start and due dates of its all problems.
+            if unit['category'] == u'vertical':
+                for component in block.get_children():
+                    # override start and due date of problem (Copy dates of vertical into problems)
+
+                    if getattr(component, 'start', False) and start:
+                            override_field_for_ccx(ccx, component, 'start', start)
```
Comment (justinabrahms) [inline lms/djangoapps/ccx/views.py]: The spacing here seems off? 8 space indents instead of 4?

Comment (jamiefolsom) [inline lms/djangoapps/ccx/views.py]: Tabs v. spaces perhaps?

File: lms/djangoapps/ccx/views.py [hunk 3]
```diff
@@ -234,6 +234,17 @@ def override_fields(parent, data, graded, earliest=None, ccx_ids_to_delete=None)
                 graded[block.format] = graded.get(block.format, 0) + 1
 
             children = unit.get('children', None)
+            # For a vertical override start and due dates of its all problems.
+            if unit['category'] == u'vertical':
+                for component in block.get_children():
```
Comment (pdpinch) [inline lms/djangoapps/ccx/views.py]: So now we're overriding due dates on all children of verticals? I think that's the safest thing to do. 

How does this interact with the performance changes edX made in https://github.com/edx/edx-platform/pull/9637 ?

**Comment by amir-qayyum-khan** [comment]
close via https://github.com/edx/edx-platform/pull/10619

**Comment by pdpinch** [comment]
Don't close this until edx#10619 is merged.

**Comment by amir-qayyum-khan** [comment]
ok

**Comment by amir-qayyum-khan** [comment]
Accepted on upstream https://github.com/edx/edx-platform/pull/10619

**Pull request closed without merging**